File: scrimage-core/src/main/scala/com/sksamuel/scrimage/composite/AlphaComposite.scala
package com.sksamuel.scrimage.composite

import java.awt.Graphics2D

import com.sksamuel.scrimage.{AwtImage, Composite}

/** @author Stephen Samuel */
class AlphaComposite(alpha: Double) extends Composite {
  def apply(src: AwtImage, applicative: AwtImage): Unit = {
    val g2 = src.awt.getGraphics.asInstanceOf[Graphics2D]
    g2.setComposite(java.awt.AlphaComposite.SrcOver.derive(alpha.toFloat))
    g2.drawImage(applicative.awt, 0, 0, null)
    g2.dispose()
  }
}
object AlphaComposite {
  def apply(alpha: Double): AlphaComposite = new AlphaComposite(alpha)
}

File: scrimage-core/src/main/java/com/sksamuel/scrimage/composite/AlphaComposite.java
package com.sksamuel.scrimage.composite;

import com.sksamuel.scrimage.AwtImage;
import com.sksamuel.scrimage.Composite;

import java.awt.*;

public class AlphaComposite implements Composite {

    private final double alpha;

    public AlphaComposite(double alpha) {
        this.alpha = alpha;
    }

    @Override
    public void apply(AwtImage src, AwtImage overlay) {
        Graphics2D g2 = (Graphics2D) src.awt.getGraphics();
        g2.setComposite(java.awt.AlphaComposite.SrcOver.derive((float) alpha));
        g2.drawImage(overlay.awt, 0, 0, null);
        g2.dispose();
    }
}